x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/action/role/PutRoleRequestTests.java

```diff
@@ -30,6 +41,18 @@ public void testValidationOfApplicationPrivileges() {
             buildRequestWithApplicationPrivilege("%*", new String[]{"all"}, new String[]{"*"}));
     }
 
+    public void testSerialization() throws IOException {
+        final PutRoleRequest original = buildRandomRequest();
+
+        final BytesStreamOutput out = new BytesStreamOutput();
+        original.writeTo(out);
+
+        final PutRoleRequest copy = new PutRoleRequest();
+        copy.readFrom(out.bytes().streamInput());
+
+        assertThat(copy.roleDescriptor(), equalTo(original.roleDescriptor()));
```

Comment:
I see that `RoleDescriptor.ApplicationPrivilege#equals` https://github.com/elastic/elasticsearch/blob/b8d7462507bae4649b22b810d5efd9efa2eae05f/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptor.java#L839 does not check for application name to be equal.
This looks to me as an oversight?